### PR TITLE
Chore/upgrade styled components2

### LINKED ISSUE #9860: Dependencies: styled-components
- update `styled-components` to `v6`

**Issues:**

- some small breaking changes can occur https://styled-components.com/releases
- we have redesign in progress, would be worth it to update it there and backport
- consider not using vendor prefixes

### PR DESCRIPTION
**Pull request by peter-sanderson**
## Description

Stuff needs to be communicated:
- [ ] You have to use `&` in case of [nested syntax](https://styled-components.com/docs/faqs#nested-syntax-handling) (fro example: `&::after`
- [ ] You have to prefix styled component with `$` (its called *transient props*) to NOT propagate them into DOM when they are `camelCase` 
image see: https://styled-components.com/docs/faqs#shouldforwardprop-is-no-longer-provided-by-default

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/9860

### CODE REVIEW

**Comment by peter-sanderson** [comment]
I got blocked by error: `_ref$theme is undefined`

image


- It appears only after styled component update
- but seems to be that some `theme` is missing deep inside of react-select component code
- It can be reproduce in SignAndVerify Form
- It is possible to `console.log` debug it by adding console logs into `/node_modules/react-select/dist/index-.......` files

image

File: yarn.lock [hunk 1]
```diff
@@ -1813,6 +1822,24 @@ __metadata:
   languageName: node
   linkType: hard
 
+"@babel/traverse@npm:^7.4.5":
```
Comment (peter-sanderson) [inline yarn.lock:1825]: can we de-duplictate this?

Comment (komret) [reply yarn.lock:1825]: Yes, just run `yarn dedupe`.

**Comment by peter-sanderson** [comment]
Closed in favor of https://github.com/trezor/trezor-suite/pull/11560

**Pull request closed without merging**